.readthedocs.yml

```diff
@@ -1,15 +1,23 @@
 ---
 # Read the Docs configuration file
 # See https://docs.readthedocs.io/en/stable/config-file/v2.html for details
+#
+# The pre_build command checks if we're building a PR. If not, check out
+# doc/releases from the main branch so it's always up to date.
 
 version: 2
 formats: []
 build:
-  image: latest
+  os: ubuntu-22.04
+  tools:
+    python: "3.8"
   apt_packages:
     - ditaa
+  jobs:
+    pre_build:
+      - git branch | head -1 | grep -qv detached && git checkout origin/main -- doc/releases || true
```

Comment:
`git branch | head -1 | grep -qv detached` pipeline can be replaced with `git symbolic-ref HEAD`.

Reply:
Also note that ` || true` would also swallow `git checkout` errors.  I would use an `if`:
```
      - if git symbolic-ref HEAD; then git checkout origin/main -- doc/releases; fi
```

Reply:
Thank you. I was looking for a cleaner solution.  What do you think about just `git symbolic-ref HEAD || git checkout origin/main -- doc/releases`?

Reply:
AFAIU you wanted `git checkout` to be run only on named branches -- if the HEAD is detached it shouldn't be run because that is a PR build?  Your proposal would do the opposite.

Reply:
> `git branch | head -1 | grep -qv detached` pipeline can be replaced with `git symbolic-ref HEAD`.

So this isn't working for some reason and I can't reproduce it.
From https://readthedocs.org/projects/ceph/builds/17666110/
```
+ git symbolic-ref HEAD
fatal: ref HEAD is not a symbolic ref
```

Reply:
I think it's because the repository doesn't have any local branches other than `main`.  From https://readthedocs.org/projects/ceph/builds/17666110/:
```
[rtd-command-info] start-time: 2022-08-09T16:57:18.200476Z, end-time: 2022-08-09T17:00:06.389954Z, duration: 168, exit-code: 0
git clone --no-single-branch --depth 50 https://github.com/ceph/ceph.git .
Cloning into '.'...
Updating files:  79% (9090/11480)
Updating files:  80% (9184/11480)
[...]
Updating files: 100% (11480/11480), done.

[rtd-command-info] start-time: 2022-08-09T17:00:06.723182Z, end-time: 2022-08-09T17:00:07.863800Z, duration: 1, exit-code: 0
git checkout --force origin/pacific
Note: switching to 'origin/pacific'.

You are in 'detached HEAD' state. You can look around, make experimental
[...]
```
So `git symbolic-ref` naturally says that it's detached (i.e. not on any local branch):
```
[rtd-command-info] start-time: 2022-08-09T17:06:11.369227Z, end-time: 2022-08-09T17:06:12.061650Z, duration: 0, exit-code: 0
bash admin/rtd-checkout-main
+ git symbolic-ref HEAD
fatal: ref HEAD is not a symbolic ref
```
Your `git branch | head -1 | grep -qv detached` would have done the same because the first line of `git branch` output would have looked like:
```
* (HEAD detached at origin/pacific)
```